byline.go

```diff
@@ -145,17 +145,15 @@ func (lr *Reader) MapStringErr(filterFn func(string) (string, error)) *Reader {
 	})
 }
 
-// Each - processing each line without changing the line
+// Each - processing each line
```

Comment:
- [ ] Consider documenting the constraints on the byte slice supplied to the "filterFn" argument:  
  - Will it be overwritten on the next call?  
    I'm not sure if you're reusing the backing array for reading each line.
  - If so, the caller must not retain it, assuming that it's immutable.